### ISSUE TITLE
Meta: Refactoring for version 0.3

### ISSUE THREAD

**Issue by andrey-kuprianov** [informalsystems/modelator#65]
This is a meta-issue to keep track of refactorings necessary for the version 0.3, to be release end of August.
We will not do a complete refactoring to fully implement #2; instead we will do a sequence of simple steps, each making the tool more reliable and easier to use.

The tasks are listed in the approximate order of their execution, but some can be done in parallel.

- [x] #53
- [x] #58 
- [x] #61
- [x] #62
- [x] #63
- #54 (postponed to v.0.4)
- #64 (postponed to v.0.4)

**Comment by andrey-kuprianov** [comment]
Closing as all tasks are either done or postponed to the next version
